server/src/test/scala/services/advpaxinfo/SplitsExportSpec.scala

```diff
@@ -0,0 +1,174 @@
+    Tuple2(PaxTypes.NonVisaNational.cleanName, Queues.NonEeaDesk)
+  )
+
+  "Looking at raw API data" >> {
+
+    "I can produce an csv export of nationalities" >> {
+      skipped("These were used to help write the exporting code for real API files which are not normally accessible")
+      val files = SplitsExport.getListOfFiles(rawZipFilesPath)
+
+      val carriers = List(
+        "AB", "NY", "AP", "IZ", "5O", "AG", "5Y", "S4", "8H", "0B", "OK", "J7", "WK", "OF", "5F", "FH", "FH",
+        "4U", "5K", "6H", "B0", "VL", "IG", "NL", "SX", "RE", "ZT", "VG", "W6", "8Z",
+        "JP", "AP", "IZ", "5O", "BM", "V3", "V4", "WX", "DE", "OU", "J7", "T3", "T4", "OF", "E9", "NY", "5F",
+        "ST", "IB", "I2", "6H", "LG", "IG", "YM", "HG", "SX", "EZ", "ZT", "HV", "OR", "WF", "WW")
+
+      val filesToInclude = files.sortBy(_.getAbsolutePath)
+      val (nations, summaries) = SplitsExport.extractSummariesFromZips(filesToInclude, carriers)
+
+      SplitsExport.writeCsvReport(nations, summaries, "/tmp/apiNats.csv")
+
+      1 !== 0
+    }
+
+    "I can combine flight summaries into csv lines" >> {
+      val summaries = List(
+        FlightSummary("BY7339", "2017-10-11", "21:00:00", "EMA", isInteractive = Option(false), Map("GBR" -> 198, "BLZ" -> 1, "LTU" -> 1)),
+        FlightSummary("BY7339", "2017-10-12", "21:00:00", "EMA", isInteractive = Option(false), Map("GBR" -> 150, "FRA" -> 20)))
+
+      val expected = List(
+        FlightSummary("BY7339", "2017-10-11", "21:00:00", "EMA", isInteractive = Option(false), Map("GBR" -> 198, "BLZ" -> 1, "LTU" -> 1, "FRA" -> 0)),
+        FlightSummary("BY7339", "2017-10-12", "21:00:00", "EMA", isInteractive = Option(false), Map("GBR" -> 150, "BLZ" -> 0, "LTU" -> 0, "FRA" -> 20)))
+
+      val (_, result) = SplitsExport.expandToFullNationalities(summaries)
+
+      result === expected
+    }
+
+    "I can get export splits from API directly to csv" >> {
+      skipped("no need to export the csv every time")
+      import akka.stream.scaladsl._
+
+      val files = SplitsExport
+        .getListOfFiles(rawZipFilesPath)
+        .filterNot(f => {
+          val fileNameParts = f.getName.split("_")
+          val date = Integer.parseInt(fileNameParts(2))
+          val tooOld = date < 161201
+          if (tooOld) println(s"ignoring old zip ${f.getName} - $date < 161201")
+          tooOld
+        })
+
+      val filePath = "/home/rich/dev/all-splits-from-api.csv"
+      val file = new File(filePath)
+      val bw = new BufferedWriter(new FileWriter(file))
+      val paxTypes = archetypes.map(_._1).mkString(",")
+      bw.write(s"flight,scheduled,origin,dest,year,month,day,$paxTypes\n")
+
+      files.sortBy(_.getName).foreach(file => {
+        val byteStringSource = FileIO.fromPath(Paths.get(file.getAbsolutePath))
+        writeSplitsFromZip(file.getName, byteStringSource, bw)
+      })
+
+      bw.close()
+
+      1 === 1
+    }
+  }
+
+  implicit val actorSystem: ActorSystem = ActorSystem("AdvPaxInfo")
```

Comment:
This will read in the application.conf - you need to user AkkaTestKit and get the system out of there.

Reply:
I'm willing to live with this for now